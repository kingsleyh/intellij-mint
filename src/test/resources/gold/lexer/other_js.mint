module Js {

  fun generateNewWallet (networkPrefix : String) : Result(Wallet.Error, Wallet) {
      `
      (() => {
        try {
          if (["T0", "M0"].indexOf(networkPrefix) === -1) {
            return new Err($Wallet_Error_InvalidNetwork)
          }

          var keyPair = generateValidKeyPair();

          var wallet = {
            publicKey: keyPair.hexPublicKey,
            wif: makeWif(keyPair.hexPrivateKey, networkPrefix),
            address: makeAddress(keyPair.hexPublicKey, networkPrefix)
          }

          return new Ok(new Record(wallet))
        } catch (e) {
          return new Err($Wallet_Error_WalletGenerationError)
        }
      })()
      `
    }

    fun encryptWallet (wallet : Wallet, password : String) : Result(Wallet.Error, EncryptedWallet) {
      `
      (() => {
        try {
          var address = wallet.address;
          var salt = all_crypto.bcrypt.genSaltSync(10)
          var hash = all_crypto.bcrypt.hashSync(password, salt)

          var walletJson = JSON.stringify(wallet);
          var bf = new all_crypto.blowfish(reverseString(hash));
          var ciphertext = ab2hexstring(bf.encode(walletJson));

          var encryptedWallet = {
                 source: "kajiki",
                 ciphertext: ciphertext,
                 address: address,
                 salt: salt
          };

          return new Ok(new Record(encryptedWallet))
        } catch (e) {
          return new Err($Wallet_Error_EncryptWalletError)
        }
      })()
      `
    }

    fun decryptWallet (encryptedWallet : EncryptedWallet, password : String) : Result(Wallet.Error, Wallet) {
      `
      (() => {
        try {
          var hash = all_crypto.bcrypt.hashSync(password, encryptedWallet.salt)
          var bf = new all_crypto.blowfish(reverseString(hash));
          var binaryCipherText = new Uint8Array(hexstring2ab(encryptedWallet.ciphertext));
          var wallet = JSON.parse(bf.decode(binaryCipherText));

          return new Ok(new Record(wallet))
        } catch (e) {
          return new Err($Wallet_Error_DecryptWalletError)
        }
      })()
      `
    }


}