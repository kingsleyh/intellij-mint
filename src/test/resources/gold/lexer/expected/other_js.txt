module ('module')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Js')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n\n  ')
fun ('fun')
WHITE_SPACE (' ')
LOWER_IDENT ('generateNewWallet')
WHITE_SPACE (' ')
( ('(')
LOWER_IDENT ('networkPrefix')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Result')
( ('(')
CAPITALISED_IDENT ('Wallet')
. ('.')
CAPITALISED_IDENT ('Error')
, (',')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Wallet')
) (')')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
STRING_LITERAL ('`\n      (() => {\n        try {\n          if (["T0", "M0"].indexOf(networkPrefix) === -1) {\n            return new Err($Wallet_Error_InvalidNetwork)\n          }\n\n          var keyPair = generateValidKeyPair();\n\n          var wallet = {\n            publicKey: keyPair.hexPublicKey,\n            wif: makeWif(keyPair.hexPrivateKey, networkPrefix),\n            address: makeAddress(keyPair.hexPublicKey, networkPrefix)\n          }\n\n          return new Ok(new Record(wallet))\n        } catch (e) {\n          return new Err($Wallet_Error_WalletGenerationError)\n        }\n      })()\n      `')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
fun ('fun')
WHITE_SPACE (' ')
LOWER_IDENT ('encryptWallet')
WHITE_SPACE (' ')
( ('(')
LOWER_IDENT ('wallet')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Wallet')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('password')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Result')
( ('(')
CAPITALISED_IDENT ('Wallet')
. ('.')
CAPITALISED_IDENT ('Error')
, (',')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('EncryptedWallet')
) (')')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
STRING_LITERAL ('`\n      (() => {\n        try {\n          var address = wallet.address;\n          var salt = all_crypto.bcrypt.genSaltSync(10)\n          var hash = all_crypto.bcrypt.hashSync(password, salt)\n\n          var walletJson = JSON.stringify(wallet);\n          var bf = new all_crypto.blowfish(reverseString(hash));\n          var ciphertext = ab2hexstring(bf.encode(walletJson));\n\n          var encryptedWallet = {\n                 source: "kajiki",\n                 ciphertext: ciphertext,\n                 address: address,\n                 salt: salt\n          };\n\n          return new Ok(new Record(encryptedWallet))\n        } catch (e) {\n          return new Err($Wallet_Error_EncryptWalletError)\n        }\n      })()\n      `')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
fun ('fun')
WHITE_SPACE (' ')
LOWER_IDENT ('decryptWallet')
WHITE_SPACE (' ')
( ('(')
LOWER_IDENT ('encryptedWallet')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('EncryptedWallet')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('password')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Result')
( ('(')
CAPITALISED_IDENT ('Wallet')
. ('.')
CAPITALISED_IDENT ('Error')
, (',')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Wallet')
) (')')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
STRING_LITERAL ('`\n      (() => {\n        try {\n          var hash = all_crypto.bcrypt.hashSync(password, encryptedWallet.salt)\n          var bf = new all_crypto.blowfish(reverseString(hash));\n          var binaryCipherText = new Uint8Array(hexstring2ab(encryptedWallet.ciphertext));\n          var wallet = JSON.parse(bf.decode(binaryCipherText));\n\n          return new Ok(new Record(wallet))\n        } catch (e) {\n          return new Err($Wallet_Error_DecryptWalletError)\n        }\n      })()\n      `')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n\n')
} ('}')